corehq/apps/custom_data_fields/tests/test_fields.py

```diff
@@ -21,6 +21,48 @@ def test_validate_required(self):
         self.assertIsNone(optional_field.validate_required('sea cucumber'))
         self.assertIsNone(optional_field.validate_required(None))
 
+    def test_validate_required_for(self):
```

Comment:
This test should be broken up into 4 individual tests, so that it gets through all the scenarios without getting stuck at an earlier assert.

Reply:
[8dd8f94](https://github.com/dimagi/commcare-hq/pull/35218/commits/8dd8f94b8bc4a4b453438d23bbee8391b8838d61)